tests/unit_tests/test_lib.py

```diff
@@ -993,3 +993,36 @@ def test_sample_external_source(run_in_tmpdir, mpi_intracomm):
         assert p1.wgt == p2.wgt
 
     openmc.lib.finalize()
+
+def test_boundingbox_consistency():
+    
+    s19 = openmc.YCylinder(0, 0, 17.7)
+    s48 = openmc.Plane(0.7071067811865476, 6.123233995736766e-17, 0.7071067811865476, 11.45)
+    s58 = openmc.Plane(0.7071067811865476, 6.123233995736766e-17, 0.7071067811865476, 14.35)
+    s62 = openmc.Plane(6.123233995736766e-17, 1.0, 6.123233995736766e-17, 1.5999999999999999)
+    s64 = openmc.Plane(6.123233995736766e-17, 1.0, 6.123233995736766e-17, -1.3)
+    s65 = openmc.Plane(-0.7071067811865475, 6.123233995736766e-17, 0.7071067811865476, 1.45)
+    s68 = openmc.Plane(-0.7071067811865475, 6.123233995736766e-17, 0.7071067811865476, -1.45)
+    s101 = openmc.YPlane(5.6, boundary_type='vacuum')
+
+    region = +s64 & -s101 & -s19 & +s48 & (+s58 | +s62 | -s64 | +s65 | -s68)
+    cell = openmc.Cell(region=region)
+
+    univ = openmc.Universe(cells=[cell])
+    geometry = openmc.Geometry(root=univ)
+    settings = openmc.Settings(particles=100, batches=100)
+    model = openmc.model.Model(geometry=geometry, settings=settings)
+    model.export_to_model_xml()
+
+    openmc.lib.init()
+    #print(cell.bounding_box)
+    py_lowerleft, py_upperright = cell.bounding_box
+    assert tuple(py_lowerleft) == (-17.7, -1.3, -17.7)
+    assert tuple(py_upperright) == (17.7, 5.6, 17.7)
+    #print(openmc.lib.cells[1].bounding_box)
+
+    cpp_lowerleft, cpp_upperright = openmc.lib.cells[1].bounding_box
+    assert tuple(cpp_lowerleft) == (-17.7, -inf, -17.7)
+    assert tuple(cpp_upperright) == (17.7, 5.6, 17.7)
+
+    openmc.lib.finalize()
```

Comment:
1. How is this topical to this PR?
2. What value is this adding beyond those already in `tests/unit_tests/test_bounding_box.py`?
3. Generally it's bad practice to have something that must always run to free resources (i.e., `finalize`) outside of a `try... finally`. `with openmc.lib.run_in_memory():` is an option. 
4. Don't use `==` [with a floating point for testing](https://en.wikipedia.org/wiki/Floating-point_arithmetic#Accuracy_problems). 

If you moved this to a separate PR I'd be ready to approve.

Reply:
@MicahGale sorry this is not relevant to this PR. This is one single part of a separate bug I'm working on. I *do* have a separate local branch called fix_2939 but.... somehow something got corrupted into my develop branch. I'll rebase and figure it out. Again, apologies